From 2106f2cbf2431328a6ad05fed3897d7902ac447b Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Mon, 26 Sep 2016 21:16:38 -0700
Subject: [PATCH] LockClock: Remove wakelock

A weather widget does not deserve a wakelock. This went wild and killed
battery, but instead of finding out why the wakelock was not released,
just remove it because having a wakelock for this does not make much sense.

Change-Id: I0e9d1560988fdb90a61d7b9a09f72bffa50cf40c
---
 AndroidManifest.xml                                   |  1 -
 .../lockclock/weather/WeatherUpdateService.java       | 11 -----------
 2 files changed, 12 deletions(-)

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index 0b1ac30..641615c 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -35,7 +35,6 @@
 
     <!-- General -->
     <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
-    <uses-permission android:name="android.permission.WAKE_LOCK" />
 
     <uses-permission android:name="lineageos.permission.ACCESS_WEATHER_MANAGER" />
     <uses-permission android:name="lineageos.permission.READ_WEATHER" />
diff --git a/src/org/lineageos/lockclock/weather/WeatherUpdateService.java b/src/org/lineageos/lockclock/weather/WeatherUpdateService.java
index 711d571..55689cd 100644
--- a/src/org/lineageos/lockclock/weather/WeatherUpdateService.java
+++ b/src/org/lineageos/lockclock/weather/WeatherUpdateService.java
@@ -32,8 +32,6 @@ import android.os.HandlerThread;
 import android.os.IBinder;
 import android.os.Looper;
 import android.os.Message;
-import android.os.PowerManager;
-import android.os.PowerManager.WakeLock;
 import android.os.SystemClock;
 import android.text.TextUtils;
 import android.util.Log;
@@ -200,7 +198,6 @@ public class WeatherUpdateService extends Service {
 
         private Handler mHandler;
         private boolean mIsProcessingWeatherUpdate = false;
-        private WakeLock mWakeLock;
         private PendingIntent mTimeoutPendingIntent;
         private int mRequestId;
         private final LineageWeatherManager mWeatherManager;
@@ -273,12 +270,6 @@ public class WeatherUpdateService extends Service {
             }
 
             mIsProcessingWeatherUpdate = true;
-            final PowerManager pm
-                    = (PowerManager) mContext.getSystemService(POWER_SERVICE);
-            mWakeLock = pm.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK, TAG);
-            mWakeLock.setReferenceCounted(false);
-            if (D) Log.v(TAG, "ACQUIRING WAKELOCK");
-            mWakeLock.acquire();
 
             WeatherLocation customWeatherLocation = null;
             if (Preferences.useCustomWeatherLocation(mContext)) {
@@ -400,8 +391,6 @@ public class WeatherUpdateService extends Service {
             finishedIntent.putExtra(EXTRA_UPDATE_CANCELLED, updateCancelled);
             mContext.sendBroadcast(finishedIntent);
 
-            if (D) Log.d(TAG, "RELEASING WAKELOCK");
-            mWakeLock.release();
             mIsProcessingWeatherUpdate = false;
             mContext.stopService(new Intent(mContext, WeatherUpdateService.class));
         }
-- 
2.17.1

